addons/mail/static/src/models/message_view.js

```diff
@@ -246,6 +247,34 @@ registerModel({
         composerViewInEditing: one('ComposerView', {
             inverse: 'messageViewInEditing',
         }),
+        /**
+         * Enriched date (contains the from, email-like)
+         */
+        dateEnriched: attr({
+            compute() {
+                if (!this.message) {
+                    return clear();
+                }
+                if (this.message.message_type === 'email' && this.message.email_from) {
+                    if (this.message.datetime) {
+                        return sprintf(
+                            this.env._t("Sent by %s on %s"),
+                            this.message.email_from,
+                            this.message.datetime
+                        );
+                    }
+                    else {
+                        return sprintf(
+                            this.env._t("Sent by %s"),
+                            this.message.email_from
+                        );
+                    }
+                }
+                return sprintf(
+                    this.env._t("Posted on %s"), this.message.datetime
+                );
```

Comment:
strange lack of consistency where existence of datetime is checked on the first case but not the second